### PR TITLE
feat: add provider function to improve build size and tree-shaking

### PR DESCRIPTION
**Pull request by osnoser1**
**What are you adding/fixing?**
A new provider function that imports only services without components and will reduce the final bundle size of the application.

Before:
image

After: 
image

**Have you added tests for your changes?**
Nop. 😞  

**Will this need documentation changes?**
Yes, I did the changes, but I'm not sure if I updated all the places.

**Does this introduce a breaking change?**
Nop, it is a non breaking change 😄 

**Other information**
Nope.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -115,6 +115,16 @@ import { NgxStripeModule } from 'ngx-stripe';
 export class AppModule {}
 ```
 
+or using the provider function on bootstrap app config
+
+```ts
+import { provideNgxStripe } from 'ngx-stripe';
+
+bootstrapApplication(AppComponent, {
+  providers: [provideNgxStripe('***your-stripe-publishable-key***')]
```
Comment (richnologies) [inline README.md:124]: I actually like this name better, but I think in the actual code you name it `provideStripe`, maybe we can change it and make it all `provideNgxStripe`

**Review by richnologies — COMMENTED** [review]
Wow @osnoser1! 

I was working on something similar, but I really like this PR. Give me some time to make some tests on my own and then we can definitely merge it.

File: README.md (hunk 1 above)
Comment (osnoser1) [reply README.md:124]: Hey @richnologies, sure 😄, I  wasn't sure about what name to use, I changed it and forgot the README.
I'll make the change 👍🏻

Comment (osnoser1) [reply README.md:124]: Done! 😄

**Comment by richnologies** [comment]
@osnoser1 thank you so much for this significant contribution. 

I'm making some final tests over the weekend and I want to improve the docs a little bit. I'll be issuing a new version `16.2.0` with this feature

Thanks again

**Comment by osnoser1** [comment]
> @osnoser1 thank you so much for this significant contribution. 
> 
> I'm making some final tests over the weekend and I want to improve the docs a little bit. I'll be issuing a new version `16.2.0` with this feature
> 
> Thanks again

Perfect @richnologies, it's a pleasure to contribute, thanks to you 😄💪🏼

**Pull request merged**